### ISSUE TITLE
async function (...) not available before definition

### ISSUE THREAD

**Issue by igisev** [rpetrich/babel-plugin-transform-async-to-promises#5]
**IN**
```javascript
fun();      // <-- TypeError: fun is not a function

function wait() {
    return new Promise(resolve => setTimeout(resolve, 1000));
}

async function fun() {
    await wait();
}
```

**OUT**
```javascript
fun();                            // <-- usage

var fun = _async(function () {    // <-- definition
    return _callIgnored(wait);
});

function wait() {
    return new Promise(function (resolve) {
        return setTimeout(resolve, 1000);
    });
}
```

**Comment by rpetrich** [comment]
Fixed in v0.5.7

**Comment by igisev** [comment]
> Fixed in v0.5.7

No, it is not
The following example shows an error

```javascript
test();      // <-- TypeError: fun is not a function

function wait() {
    return new Promise(resolve => setTimeout(resolve, 1000));
}

let a;
let b;
let c;

function f1() {}
function f2() {}

let d;
let e;
let f;

// ...

function f3() {}
function f4() {}

// ...

async function test() {
    await wait();
    console.log(1);
}
```

I think the best solution in this case is to move the transpiled async-functions `async function f() { ... }` to the top of the current code-scope `{ ... }`

**Comment by igisev** [comment]
hmm, v0.6.0 is released
but what about this bug?
